functions/src/modules/gmail.functions.ts

```diff
@@ -0,0 +1,273 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import * as Google from 'googleapis';
+import * as querystring from 'querystring';
+import { Message } from 'firebase-functions/lib/providers/pubsub';
+
+const db = admin.database();
+
+export const oauth2init = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    const { client_key, secret, redirect } = functions.config().gmail;
+    const oauth2Client = new Google.google.auth.OAuth2(
+      client_key,
+      secret,
+      redirect
+    );
+    const authURL = oauth2Client.generateAuthUrl({
```

Comment:
Can we specify what particular account do we want to authorize to? We have coordinator gmail account in the config, and we actually want to get access to it, not any other. Please research, I remember there should be a means for that.

Reply:
Hmm. Okay. will research

Reply:
The owner of the account still needs to give permission though. Such as in this flow oauth2 flow. You mean that to find a way where the owner of the account does not have to go through the oauth consent process?

Reply:
No, the consent process is okay and needed.

Found the docs: https://developers.google.com/identity/protocols/OAuth2WebServer (search for `login_hint`).

Reply:
Done.